src/librustc/ty/context.rs

```diff
@@ -387,8 +387,9 @@ pub struct TypeckTables<'tcx> {
     cast_kinds: ItemLocalMap<ty::cast::CastKind>,
 
     /// Set of trait imports actually used in the method resolution.
-    /// This is used for warning unused imports.
-    pub used_trait_imports: DefIdSet,
+    /// This is used for warning unused imports. During type
+    /// checking, this field should not be cloned.
```

Comment:
It'd be nice to say a *bit* more here -- maybe:

"During type checker, this `Rc` should not be cloned: it must always have a ref-count of 1 so that we can insert things into the set mutably."